doc/sphinx/proof-engine/tactics.rst

```diff
@@ -460,16 +460,22 @@ Examples:
 Occurrence sets and occurrence clauses
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-An occurrence clause is a modifier to some tactics that obeys the
+An occurrence clause (:n:`in @in_clause`) is a modifier to some tactics that obeys the
 following syntax:
 
-  .. productionlist:: coq
-     occurrence_clause : in `goal_occurrences`
-     goal_occurrences : [`ident` [`at_occurrences`], ... , `ident` [`at_occurrences`] [|- [* [`at_occurrences`]]]]
-                      : * |- [* [`at_occurrences`]]
-                      : *
-     at_occurrences : at `occurrences`
-     occurrences     : [-] `num` ... `num`
+  .. insertprodn in_clause hypident
+
+  .. prodn::
+     in_clause ::= @in_clause
+     | * {? at @occs_nums }
+     | * |- {? @concl_occ }
+     | {*, @hypident_occ } |- {? @concl_occ }
+     | {*, @hypident_occ }
+     concl_occ ::= * {? at @occs_nums }
+     hypident_occ ::= @hypident {? at @occs_nums }
+     hypident ::= @ident
+     | ( type of @ident )
+     | ( value of @ident )
```

Comment:
The definitions of `occs_nums` (that I would like to rename `occurrences`), `num_or_var` and `int_or_var` should be moved here.

Reply:
FWIW occs_nums is used in 11 places in orderedGrammar.  I'll move this later since I'm going to back out this part of the PR.

Reply:
Then what name do you want to give to the existing `occurrences`?

```

occurrences: [
| LIST1 int
| ident
]
```

Reply:
Not sure what it is for (i.e. in which context it is used).

Reply:
The same tactics I mentioned in other comments:

```
| "rewrite" "*" orient uconstr "in" hyp "at" occurrences by_arg_tac
| "rewrite" "*" orient uconstr "at" occurrences "in" hyp by_arg_tac
| "rewrite" "*" orient uconstr "at" occurrences by_arg_tac
| "setoid_rewrite" orient glob_constr_with_bindings "at" occurrences
| "setoid_rewrite" orient glob_constr_with_bindings "at" occurrences "in" hyp
| "setoid_rewrite" orient glob_constr_with_bindings "in" hyp "at" occurrences
```